tests/integration/state_machine/test_state_machine.py

```diff
@@ -1,88 +1,66 @@
 import time
+from collections import deque
 from threading import Thread
 
+import pytest
+
 from isar.models.mission import Mission
 from isar.services.utilities.scheduling_utilities import SchedulingUtilities
 from isar.state_machine.state_machine import StateMachine, States, main
 from isar.storage.storage_interface import StorageInterface
 from robot_interface.models.mission import DriveToPose, MissionStatus, Step
 from robot_interface.robot_interface import RobotInterface
-from tests.mocks.blob_storage import StorageMock
 from tests.test_utilities.mock_interface.mock_robot_interface import MockRobot
 from tests.test_utilities.mock_models.mock_robot_variables import mock_pose
 from tests.test_utilities.mock_models.mock_step import MockStep
 
 
-def start_state_machine_in_thread(injector) -> StateMachine:
-    state_machine_thread: Thread = Thread(target=main, args=[injector])
-    state_machine_thread.daemon = True
-    state_machine_thread.start()
-    state_machine: StateMachine = injector.get(StateMachine)
-    return state_machine
+class StateMachineThread(object):
+    def __init__(self, injector) -> None:
+        self.injector = injector
+        injector.binder.bind(RobotInterface, to=MockRobot())
+        self.state_machine = injector.get(StateMachine)
+        self._thread = Thread(target=main, args=[injector])
+        self._thread.daemon = True
+        self._thread.start()
+
 
+@pytest.fixture
+def state_machine_thread(injector) -> StateMachineThread:
+    return StateMachineThread(injector)
 
-def test_state_machine(injector, mocker):
-    injector.binder.bind(RobotInterface, to=MockRobot())
-    state_machine: StateMachine = start_state_machine_in_thread(injector)
 
+def test_state_machine_transitions(injector, mocker, state_machine_thread):
     step: Step = DriveToPose(pose=mock_pose())
     mission: Mission = Mission([step])
 
     scheduling_utilities: SchedulingUtilities = injector.get(SchedulingUtilities)
     message, _ = scheduling_utilities.start_mission(mission=mission)
     assert message.started
 
-    time.sleep(1)
-    assert state_machine.status.current_state is States.Monitor
-
     mocker.patch.object(
         MockRobot, "mission_status", return_value=MissionStatus.Completed
     )
     time.sleep(1)
```

Comment:
Since the `MockRobot` implementation [here](https://github.com/equinor/isar/blob/main/tests/test_utilities/mock_interface/mock_robot_interface.py) returns `MissionStatus.InProgress` as default because of these timing based tests.

Should the default mock value for `mission_status` be changed to `MissionStatus.Completed` now that we are no longer relying on the timing based tests?

Reply:
That would remove the need for this patch.

I don't think there are other tests than the timing based ones that rely on this default value.

Reply:
Yeah, I agreed. Will change to MissionStatus.Completed